File: src/main/java/my/maryvk/maryvkweb/domain/Relation.java
package my.maryvk.maryvkweb.domain;

import lombok.*;

import javax.persistence.*;
import java.io.Serializable;
import java.time.LocalDateTime;

@Entity
@Data @Builder @AllArgsConstructor @NoArgsConstructor
public class Relation implements Serializable {

    @Id @GeneratedValue
    private Long id;

    private Integer ownerId;
    private Integer targetId;
    private RelationType relationType;

    public RelationChange createRelationChange(boolean isAppeared) {
        return RelationChange.builder()
                .time(LocalDateTime.now())
                .ownerId(ownerId)
                .targetId(targetId)
                .relationType(relationType)
                .isAppeared(isAppeared)
                .build();
    }
}

File: src/main/java/my/maryvk/maryvkweb/domain/Relation.kt
package my.maryvk.maryvkweb.domain

import java.time.LocalDateTime
import javax.persistence.Entity
import javax.persistence.GeneratedValue
import javax.persistence.Id

@Entity data class Relation(
        @Id @GeneratedValue var id: Long?,
        var ownerId: Int?,
        var targetId: Int?,
        var relationType: RelationType?
) {
    fun createRelationChange(isAppeared: Boolean) =
            RelationChange(null, LocalDateTime.now(), ownerId, targetId, relationType, isAppeared)
}